packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.Props.ts

```diff
@@ -31,6 +32,16 @@ export interface ISearchBoxProps extends React.InputHTMLAttributes<HTMLInputElem
   onSearch?: (newValue: any) => void;
 
   /**
+   * Callback executed when the user clears the search box by either clicking 'X' or hitting escape.
+   */
+  onClear?: IEventHandler<any>;
```

Comment:
This pattern feels a bit unpredictable. Normally, you can just receive callbacks when things occur, and do your own business. In this case, if you do not call the default method, you end up not doing the default behavior. For example, if you pass in `onClear` and you never call the default, then state doesn't get set. This doesn't follow the normal pattern of event handlers, which expect you to call `ev.preventDefault` to avoid the default behavior.

I think we should probably follow precedence that html eventing provides... call the callback, check ev.defaultPrevented or whatever to determine if the default behavior occurs. (or just do the default always...)

Reply:
Okay, so default to the normal behavior happening unless the developer supplies ev.preventDefault, rather than requiring them to perform the default behavior on their own?

Reply:
for `ev.preventDefault`, do you want to expose the "raw" events? Like, passing the React.Event object to the caller? Or have our own notion of a component event? 

Are there cases when you want to prevent browser default behavior but not "our" default behavior?

Reply:
hmmm, i'm not able to stop the default behavior from happening. Here's escape:
```js
 case KeyCodes.escape:
        this.props.onEscape && this.props.onEscape(ev);
        this._onClear(ev);
        break;
```

and then the onEscape prop:

```
 onEscape={ (ev) => {
            console.log('Custom onEscape Called');
            ev.stopPropagation();
            ev.preventDefault();
          } }
```

But onClear is still called.